packages/frontend/src/features/metricsCatalog/components/MetricPeekModal.tsx

```diff
@@ -84,21 +88,48 @@ export const MetricPeekModal: FC<Props> = ({ opened, onClose }) => {
         }
     }, [comparisonType, metricQuery.isSuccess, metricQuery.data]);
 
+    const [timeDimensionOverride, setTimeDimensionOverride] = useState<
+        TimeDimensionConfig | undefined
+    >();
     const metricResultsQuery = useRunMetricExplorerQuery({
         projectUuid,
         exploreName: tableName,
         metricName,
         comparison: comparisonParams,
         dateRange,
+        timeDimensionOverride,
     });
 
+    const timeDimensionBaseField: TimeDimensionConfig | undefined =
```

Comment:
this is so I can get the field's field name, interval and table so some things can be manipulated: 
- field 
- interval/granularity
the `table` is so that we can query from joined fields